### PR TITLE
Hide Deprecated Templates from Templates Menu

### PR DESCRIPTION
**Pull request by Dala0**
Adds a 'Deprecated' value as a Boolean to template records and allows it to be set to on or off through the Web based interface.

The value now shows up in the templates list as a new column.

Any templates with 'Deprecated' set to true will now not show up in the 'Project Template' drop down menu when a user starts project creation.

Tested on my local machine.

@fredex42

### CODE REVIEW

**Comment by Dala0** [comment]
[Hiding older templates from templates list](https://trello.com/c/NLqPif6I/2825-hiding-older-templates-from-templates-list)

File: frontend/app/GeneralListComponent.jsx [hunk 1]
```diff
@@ -92,6 +92,15 @@ class GeneralListComponent extends React.Component {
         };
     }
 
+    static boolColumn(name, key) {
+        return {
+            header: name,
+            key: key,
+            headerProps: { className: 'dashboardheader'},
+            render: value=><span>{ String( value ) }</span>
```
Comment (fredex42) [inline frontend/app/GeneralListComponent.jsx]: would be nice to show as an icon, but this is fine for the time being :)

File: frontend/app/multistep/projecttemplate/SummaryComponent.jsx [hunk 2]
```diff
@@ -35,6 +36,10 @@ class SummaryComponent extends React.Component {
                 <td>Project type</td>
                 <td id="projectType"><ProjectTypeView entryId={this.props.projectType}/></td>
             </tr>
+            <tr>
+                <td>Deprecated</td>
+                <td id="deprecated"><span style={{display: this.props.deprecated ? "inline-block":"none"}}>Yes</span><span style={{display: !this.props.deprecated ? "inline-block":"none"}}>No</span></td>
```
[the remark this answers is not in the document]
Comment (Dala0) [reply frontend/app/multistep/projecttemplate/SummaryComponent.jsx]: Done.

**Comment by fredex42** [comment]
@Dala0 Looks good to me, pls run it up on the dev environment to check that it works as intended before merging?

We won't deploy to live before new year, but can merge it beforehand

**Pull request merged**